hudi-common/src/main/java/org/apache/hudi/common/model/HoodieKey.java

```diff
@@ -53,6 +53,9 @@ public void setPartitionPath(String partitionPath) {
   }
 
   public String getPartitionPath() {
+    if (partitionPath.trim().startsWith("/")) {
```

Comment:
this might be costly to do it for every record key. probably we should do it elsewhere once. 
Also in general, mostly everyone uses out of the box key generators and hence this was not an issue. this might be a problem only when you use your own custom key gen. 
Curious to know if none of the built in key generators worked for u?

Reply:
@nsivabalan , I'm using the Java Client and not Spark. I see that all key generators extend directly or indirectly the Build-in key gen [link](https://github.com/apache/hudi/blob/master/hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/keygen/BuiltinKeyGenerator.java) which in turn implements SparkKeyGeneratorInterface. 

**I'm using the Java Client and I want my implementation simple and without any Spark dependency.**

As for  _probably we should do it elsewhere once._   - I understand the performance impact and is not ideal. Do you have a suggestion as to where else the validation would be more suitable?